build/teamcity/cockroach/ci/tests/unit_tests_impl.sh

```diff
@@ -10,9 +10,9 @@ bazel build //pkg/cmd/bazci --config=ci
 
 EXTRA_PARAMS=""
 
-if tc_release_branch; then
-  # enable up to 2 retries (3 attempts, worst-case) per test executable to report flakes but only on release branches (i.e., not staging)
-  EXTRA_PARAMS=" --flaky_test_attempts=3" 
+if tc_release_branch || tc_bors_branch; then
+  # enable up to 1 retry (2 attempts, worst-case) per test executable to report flakes but only on release branches and staging.
+  EXTRA_PARAMS=" --flaky_test_attempts=2" 
```

Comment:
I *think* that the 4/5 successful tests would be repeated in the case you outline. I don't think Bazel is clever enough to tell the test run to only re-run the failed tests in this case. (Maybe it is? It would have to put together an appropriate regex for the test filter.)